### PR TITLE
MTC-799 injected InstanceId to create different rabbitmq queues for r…

### PR DESCRIPTION
**Pull request by Waqas-Zamir**
…ates

### CODE REVIEW

**Comment by Waqas-Zamir** [comment]
@andrey-anokhin 
3. I couldn't get because SettingsChanged consumer has `IsDurable = true`, do you want me to change it to false and create 1 queue per instance?
2. Do you prefer to use `Env` variable name to pass instance id? I tried to align InstanceId approach which is used in Lego services (e.g. Donut)

**Comment by andrey-anokhin** [comment]
@Waqas-Zamir 
2. You can name this variable InstanceId or Env, but it should be environment variable, not parameter in settings. Or you need to add ability to override value in settings from env variable.
3. Yes, let's change it to false and create 1 queue per instance

**Comment by Waqas-Zamir** [comment]
fixed

File: src/MarginTrading.CommissionService/Startup.cs [hunk 1]
```diff
@@ -76,10 +76,16 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
                 var builder = new ContainerBuilder();
                 var appSettings = Configuration.LoadSettings<AppSettings>(
-                    throwExceptionOnCheckError: !Configuration.NotThrowExceptionsOnServiceValidation());
+                    throwExceptionOnCheckError: !Configuration.NotThrowExceptionsOnServiceValidation())
+                    .Nested(s =>
+                    {
+                        s.CommissionService.InstanceId = Configuration.InstanceId();
+
+                        // Generating new instance id guid if it is not specified through configurations
+                        s.CommissionService.InstanceId = s.CommissionService.InstanceId ?? Guid.NewGuid().ToString("N");
```
Comment (andrey-anokhin) [inline src/MarginTrading.CommissionService/Startup.cs]: A little bit complicated ) Why not s.CommissionService.InstanceId = Configuration.InstanceId() ?? Guid.NewGuid().ToString("N"); ?

**Pull request merged**